src/vanilla/ClientModelExtensions.cs

```diff
@@ -714,14 +714,15 @@ private static string ContructMapperForCompositeType(this CompositeType composit
                 {
                     var prop = composedPropertyList[i];
                     var serializedPropertyName = prop.SerializedName;
+                    string propertyName = ModifyString(prop.Name);

```

Comment:
Curious: Although this would be the fix for now, but seems like we are doing too much modification that supposed to be already taken care by generic methods. 

I'd argue that what the reason of `prop.Name` not being the one we want and what do we need to fix this manually here?

Reply:
Also, It'd be great if you can add unit test for this scenario.